eng/targets/ResolveReferences.targets

```diff
@@ -216,16 +216,16 @@
         Condition=" '$(CompileUsingReferenceAssemblies)' != false AND '$(TargetFramework)' == '$(DefaultNetCoreTargetFramework)' ">true</_CompileTfmUsingReferenceAssemblies>
   </PropertyGroup>
   <!--
-    If we have a ref/ assembly from Extensions for a package, use that when compiling. The build-only reference to
-    Microsoft.Internal.Extensions.Refs ensures package is installed and $(MicrosoftInternalExtensionsRefsPath) is set.
+    If we have a ref/ assembly from dotnet/runtime for an Extension package, use that when compiling but do not reference its assemblies.
   -->
   <ItemGroup
       Condition=" $(_CompileTfmUsingReferenceAssemblies) OR ('$(IsTargetingPackBuilding)' != 'false' AND '$(MSBuildProjectName)' == 'Microsoft.AspNetCore.App.Ref') ">
-    <PackageReference Include="Microsoft.Internal.Extensions.Refs"
-        Version="$(MicrosoftInternalExtensionsRefsPackageVersion)"
+    <PackageReference Include="Microsoft.Extensions.Internal.Transport"
+        Version="$(MicrosoftExtensionsInternalTransportPackageVersion)"
         IsImplicitlyDefined="true"
-        IncludeAssets="Build"
-        PrivateAssets="All" />
+        ExcludeAssets="runtime"
```

Comment:
This seems to resolve assemblies from /ref instead of /lib

Reply:
Can you try with today's build and see if that's resolved?

Reply:
I spot checked 5.0.0-preview.3-runtime.20181.3 and it seems to fix this as well as docs and the public key.

Reply:
The comment here was actually a question. How do I get the reference resolution to *only* resolve from ref/ ? Specifically, we only want to compile against the ref assemblies from the transport and we want to avoid compiling against the implementation assemblies. I think `ExcludeAssets="runtime"` did the trick when I examined the binlog but I'm not sure if this is a well known mechanism or it was just a fluke.

Reply:
Ah, I see.  Yes this is officially supported.  We've used this mechanism since .NETCore 1.x.

Reply:
See https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets But, it may not work correctly unless you update `@(PackageReference)` items for the implementation projects to include `ExcludeAssets="compile"`. May also be safer to include specific asset types e.g. `IncludeAssets="compile"` here.

The other option is to exclude all assets but use `$(PkgMicrsoft_Extensions_Internal_Transport)` in a later target to update reference paths, as we do in the 'release/3.1' branch: https://github.com/dotnet/aspnetcore/blob/ffab087b73506f6622eff69e087efcb557efeef4/eng/targets/ResolveReferences.targets#L293-L298

Reply:
> But, it may not work correctly unless you update @(PackageReference) items for the implementation projects to include ExcludeAssets="compile".

Why would you do that?  If you don't want the transport package to be used, don't use it.  If it's showing up transitively then mark it with PrivateAssets.  I don't see why we'd exclude compile from a package that's sole purpose is to provide the reference assemblies.

Reply:
I said "the implementation projects", not this one. If something depends on Microsoft.Extensions.Options and this target, shouldn't it compile against this one but run against whatever's in the Microsoft.Extensions.Options assemblies? If we're specific about what's included from both, conflicts won't show up.

Reply:
I see.  Because you're trying to build your libraries against references in ref-pack but still ship packages which have package dependencies that don't depend on the shared framework.

Be careful about marking the PackageReferences as ExcludeAssets="compile" in shipping pack'ed projects as that marking will persist in the nuspec of the package.

Reply:
> Be careful about marking the PackageReferences as ExcludeAssets="compile" in shipping pack'ed projects as that marking will persist in the nuspec of the package.

Good point. What do you recommend instead to ensure we never see conflicts?

Reply:
For this PR don't change things.  You can ExcludeAssets all on the transport package and just switch to using $(PkgMicrsoft_Extensions_Internal_Transport).

If you're missing some fixes from `release/3.1` then bring those over separately.  I don't think anything about how were sharing the transport package should upset/change the way you build aspnetcore.

As an aside, have you considered stopping producing packages for things in the shared framework?  That would reduce your complexity here.

Reply:
> As an aside, have you considered stopping producing packages for things in the shared framework? That would reduce your complexity here.

Unfortunately most of the packages in this repo that ship in both sharedfx and packages are for Blazor WASM which runs outside of the shared framework, so I don't think we can stop doing that.